x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -506,12 +506,12 @@ private static List<Map.Entry<String, Supplier<Object>>> ulongCases(BigInteger m
 
         // Big values, greater than Long.MAX_VALUE
         BigInteger lower2 = min.max(BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE));
-        BigInteger upper2 = max.min(BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.valueOf(Integer.MAX_VALUE)));
+        BigInteger upper2 = max.min(MAX_UNSIGNED_LONG);
         if (lower2.compareTo(upper2) < 0) {
             cases.add(
                 Map.entry(
                     "<big unsigned long>",
-                    () -> BigInteger.valueOf(ESTestCase.randomLongBetween(lower2.longValue(), upper2.longValue()))
+                    () -> lower2.add(BigInteger.valueOf(ESTestCase.randomLongBetween(0, upper2.subtract(lower2).longValue())))
```

Comment:
nit: Maybe it's worth adding a comment here that explains why the arguments to `randomLongBetween` will always be valid.

(`upper2-lower2` must fit into a long, so at most `2^63-1`. The biggest possible difference is `MAX_UNSIGNED_LONG - (Long.MAX_VALUE + 1) = 2^64-1 - (2^63-1+1) = 2^63-1 = Long.MAX_VALUE`. Maybe that's super obvious to others, but took me a moment to make sure here's no accidental off-by-one error.)

Reply:
I agree with Alex.  This needs more comments, especially considering this is like the third time we've messed this up.  Frankly, this should probably move to `ESTestCase` as `randomBigIntBetween` or `randomUnsignedLongBetween`